### PR TITLE
Fix #20925, cp not preserving permissions

### LINKED ISSUE #20925: cp() does not retain source permissions on mac / linux.
If you copy any file that has execute permission, the execute permission gets removed. Execute the following command.
`cp("/bin/ls", "/tmp/ls")`

The file `/tmp/ls` will no more have execute permission.

### PR DESCRIPTION
**Pull request by tkelman**
Not positive whether this should be in the `try` or the `finally` of this function, or maybe made part of `cp` instead of `sendfile`.

### CODE REVIEW

File: base/file.jl [hunk 1]
```diff
@@ -515,7 +515,7 @@ function sendfile(src::AbstractString, dst::AbstractString)
         src_file = open(src, JL_O_RDONLY)
         src_open = true
         dst_file = open(dst, JL_O_CREAT | JL_O_TRUNC | JL_O_WRONLY,
-             S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP| S_IROTH | S_IWOTH)
+             S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH)
```
Comment (vtjnash) [inline base/file.jl]: Looks like in other cases we've been making this a `mode::Unsigned = 0o666` kw-arg, which also seems good

**Review by vtjnash — CHANGES_REQUESTED** [review]
I think I would rather have this be handled via a `mode` argument to `sendfile` to set the initial permissions. And then we might consider adding a kw-arg to `cp` also to enable `preserve_mode` (default false) to do the `chmod` call.

It's incorrect to call this "preserve permissions", since we aren't actually preserving the ACL (for posix on the local filesystem, that would – at a minimum – require also changing the uid and gid and sometimes fiddling with the sticky bits.

**Comment by tkelman** [comment]
> `preserve_mode` (default false)

would be less surprising to default to being consistent with coreutils `cp`

**Comment by vtjnash** [comment]
> would be less surprising to default to being consistent with coreutils cp

Agreed. "If the destination file does not exist, the mode of the source file is used as modified by the file mode creation mask" (e.g. I would suggest passing this as the `mode` argument I mentioned above, not via chmod afterwards)

**Pull request closed without merging**